### ISSUE TITLE
[BUG] Diff on empty series causes cuLaunchKernel errors.

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#3433]
**Describe the bug**
 Diff on empty series causes CUDA_ERROR_INVALID_VALUE. This causes bugs downstream in workflows with `dask-cudf` where we can have empty partitions. 

**Steps/Code to reproduce bug**
```python
import cudf

cudf_ser = cudf.Series(data=None,dtype='int32')
cudf_ser.diff(periods=1)
```

Error:

```
ERROR:Call to cuLaunchKernel results in CUDA_ERROR_INVALID_VALUE
---------------------------------------------------------------------------
CudaAPIError                              Traceback (most recent call last)
<ipython-input-2-bff68bc014c9> in <module>
      2 
      3 cudf_ser = cudf.Series(data=None,dtype='int32')
----> 4 cudf_ser.diff(periods=1)

/raid/vjawa/conda_env/conda_installation/envs/ucx_10_1/lib/python3.7/site-packages/cudf/core/series.py in diff(self, periods)
   2448         output_dary = rmm.device_array_like(input_dary)
   2449         cudautils.gpu_diff.forall(output_dary.size)(
-> 2450             input_dary, output_dary, periods
   2451         )
   2452         return Series(output_dary, name=self.name, index=self.index)

/raid/vjawa/conda_env/conda_installation/envs/ucx_10_1/lib/python3.7/site-packages/numba/cuda/compiler.py in __call__(self, *args)
    273 
    274         return kernel.configure(blkct, tpb, stream=self.stream,
--> 275                                 sharedmem=self.sharedmem)(*args)
    276 
    277     def _compute_thread_per_block(self, kernel):

/raid/vjawa/conda_env/conda_installation/envs/ucx_10_1/lib/python3.7/site-packages/numba/cuda/compiler.py in __call__(self, *args, **kwargs)
    535                           blockdim=self.blockdim,
    536                           stream=self.stream,
--> 537                           sharedmem=self.sharedmem)
    538 
    539     def bind(self):

/raid/vjawa/conda_env/conda_installation/envs/ucx_10_1/lib/python3.7/site-packages/numba/cuda/compiler.py in _kernel_call(self, args, griddim, blockdim, stream, sharedmem)
    609                                    sharedmem=sharedmem)
    610         # Invoke kernel
--> 611         cu_func(*kernelargs)
    612 
    613         if self.debug:

/raid/vjawa/conda_env/conda_installation/envs/ucx_10_1/lib/python3.7/site-packages/numba/cuda/cudadrv/driver.py in __call__(self, *args)
   1576 
   1577         launch_kernel(self.handle, self.griddim, self.blockdim,
-> 1578                       self.sharedmem, streamhandle, args)
   1579 
   1580     @property

/raid/vjawa/conda_env/conda_installation/envs/ucx_10_1/lib/python3.7/site-packages/numba/cuda/cudadrv/driver.py in launch_kernel(cufunc_handle, griddim, blockdim, sharedmem, hstream, args)
   1620                           hstream,
   1621                           params,
-> 1622                           None)
   1623 
   1624 

/raid/vjawa/conda_env/conda_installation/envs/ucx_10_1/lib/python3.7/site-packages/numba/cuda/cudadrv/driver.py in safe_cuda_api_call(*args)
    292             _logger.debug('call driver api: %s', libfn.__name__)
    293             retcode = libfn(*args)
--> 294             self._check_error(fname, retcode)
    295         return safe_cuda_api_call
    296 

/raid/vjawa/conda_env/conda_installation/envs/ucx_10_1/lib/python3.7/site-packages/numba/cuda/cudadrv/driver.py in _check_error(self, fname, retcode)
    327                     _logger.critical(msg, _getpid(), self.pid)
    328                     raise CudaDriverError("CUDA initialized before forking")
--> 329             raise CudaAPIError(retcode, msg)
    330 
    331     def get_device(self, devnum=0):

CudaAPIError: [1] Call to cuLaunchKernel results in CUDA_ERROR_INVALID_VALUE
```

**Expected behavior**

I would expect it to return a empty series like `pandas`.
```
import pandas as pd

pd_ser = pd.Series(data=None,dtype='int32')
pd_ser.diff(periods=1)
```

```
Series([], dtype: float64)
```

**Environment overview (please complete the following information)**
 - Method of cuDF install: conda
```
cudf                      0.11.0a191119         py37_3035    rapidsai-nightly
dask-cudf                 0.11.0a191119         py37_3035    rapidsai-nightly
libcudf                   0.11.0a191119     cuda10.1_2997    rapidsai-nightly
```

**Additional Context**

The workaround currently i am using is to filter out the  empty frames at dask-cudf level.

###### From taxi notebook
```
def drop_empty_partitions(df):
    lengths = df.map_partitions(len).compute()
    nonempty = [length > 0 for length in lengths]
    return df.partitions[nonempty]
```

**Comment by beckernick** [comment]
We can resolve this by guarding from the 0 size array. We can either wrap it into a helper function or add it to the Series method.

Helper function approach: https://github.com/rapidsai/cudf/blob/a49a20aa3c4cceb034fdf5035534a42022a6a1ec/python/cudf/cudf/utils/cudautils.py#L513-L517

For the series method, we'd just need to do the guard there.

**Comment by kkraus14** [comment]
Fixed by #3435
